lib/tasks/compare_xml_collections.rake

```diff
@@ -0,0 +1,121 @@
+    back_to_native = cmr_client.translate_collection(umm_json, "application/vnd.nasa.cmr.umm+json;version=#{args.version}", "application/#{args.format}+xml", skip_validation=true )
+    back_to_native.success? ? puts("successful translation to native format \n\n") : abort("Native format translation failure \n\n")
+    native_converted_hash = Hash.from_xml(back_to_native.body)
+    native_converted_xml = back_to_native.body
+
+    # nokogiri output
+    nokogiri_original = Nokogiri::XML(native_original_xml) { |config| config.strict.noblanks }
+    nokogiri_converted = Nokogiri::XML(native_converted_xml) { |config| config.strict.noblanks }
+
+    ignored_paths = Array.new
+
+    nokogiri_original.diff(nokogiri_converted, {:added => true, :removed => true}) do |change,node|
+      split_path = node.parent.path.split('[')
+      if node.parent.path.include?('[') && !ignored_paths.include?(split_path[0])
+        ignored_paths << split_path[0]
+        array_comparison(split_path[0], native_original_hash, native_converted_hash).each do |item|
+          puts("#{item[0]}: #{item[1]}".ljust(60) + item[2]) if args.disp.eql? 'show'
+          puts("#{item[0]}: ". + item[2]) if args.disp.eql? 'hide'
+        end
+      elsif !ignored_paths.include?(split_path[0]) && !path_leads_to_list?(node.parent.path, native_original_hash, native_converted_hash)
+        puts("#{change}: #{node.to_xml}".ljust(60) + node.parent.path) if args.disp.eql? 'show'
+        puts("#{change}: ". + node.parent.path) if args.disp.eql? 'hide'
+      end
+    end
+  end
+
+  def path_leads_to_list?(path, org_hash, conv_hash)
+    # this method takes a path string (and the full original and converted hashes) and outputs true if the path string contains a list; else false
+    org_hash = hash_navigation(path, org_hash)
+    conv_hash = hash_navigation(path, conv_hash)
+
+    if path.include?("[") && path.include?("]")
+      bool = true
+    elsif org_hash.is_a?(Hash) && conv_hash.is_a?(Hash)
+      # the number of keys must be 1 because all arrays in echo10, dif10, and iso19115 are tagged similar to:
+      # <Contacts><Contact>contact</Contact></Contacts> and so all array-containing tags will be the plural
+      # of the array name. This clause serves to identify array-containing tags when their paths aren't properly
+      # displayed by nokogiri
+      bool = true if org_hash.keys.length == 1 && org_hash[org_hash.keys[0]].is_a?(Array)
+      bool = true if conv_hash.keys.length == 1 && conv_hash[conv_hash.keys[0]].is_a?(Array)
+    elsif org_hash.is_a?(Array) || conv_hash.is_a?(Array)
+      bool = true
+    else
+      bool = false
+    end
+    bool
+  end
+
+  def hash_navigation(path, hash)
+    # Passed a path string and the hash being navigated. This method parses the path string and
+    # returns the hash at the end of the path
+    path.split('/').each do |key|
+      if hash.key?(key) && hash.is_a?(Hash)
+        hash = hash[key]
+      elsif hash.is_a? Array
```

Comment:
If `hash` could be an array, I think the comment needs to be updated, and also the parameter might need to be renamed?
Also, above and below, you're using `is_a?` with parens, but not here. They should all be consistent.